src/main/java/com/sourcery/gymapp/backend/workout/mapper/RoutineMapper.java

```diff
@@ -0,0 +1,30 @@
+package com.sourcery.gymapp.backend.workout.mapper;
+
+import com.sourcery.gymapp.backend.workout.dto.CreateRoutineDto;
+import com.sourcery.gymapp.backend.workout.dto.ResponseRoutineDto;
+import com.sourcery.gymapp.backend.workout.model.Routine;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Component
+public class RoutineMapper {
+    public ResponseRoutineDto toDto(Routine routine) {
+        return new ResponseRoutineDto(
+            routine.getId(),
+            routine.getName(),
+            routine.getDescription(),
+            routine.getCreatedAt()
+        );
+    }
+
+    public Routine toEntity(CreateRoutineDto routineDto, UUID userId) {
```

Comment:
Same here
minor: isn't it better to name it entity? So you briefly understand it will be parsing dto?

so, public Routine toEntity(CreateRoutineDto dto, UUID userId) {

(not sure about this comment though)

Reply:
Also if you only need userId which is current user, u don't need to pass userId to mapper, you can get it from currentAuditor service

Reply:
But then I'd have to to set the user in the service layer.
The currentUserId is taken from the auditor and then passed into the mapper to set it there